js/renderers/ReferenceControl.ts

```diff
@@ -0,0 +1,27 @@
+/// <reference path="../../typings/angularjs/angular.d.ts"/>
+/// <reference path="../services.ts"/>
+
+class ReferenceControl implements JSONForms.IRenderer {
+
+    priority: number = 2;
+
+    constructor(private pathResolver: JSONForms.IPathResolver) {}
+
+    render(element:IUISchemaElement, schema:SchemaElement, schemaPath:string, dataProvider:JSONForms.IDataProvider) {
+        var control = new JSONForms.ControlRenderDescription(dataProvider.data, schemaPath, element.label);
+        var normalizedPath = this.pathResolver.toInstancePath(schemaPath);
+        var linkText = element['label'] ? element['label'] : "Go to " + control.label;
```

Comment:
In regards to my other note this could be changed to ```var linkText = element['href']['label'] ? element['href']['label'] : "Go to " + control.label;```